pkg/dmesg/commands.go

```diff
@@ -0,0 +1,146 @@
+package dmesg
+
+import (
+	"context"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/leptonai/gpud/pkg/process"
+)
+
+type Commands struct {
+	WatchCommands [][]string
+	TailCommands  [][]string
+	ParseTimeFunc func(line []byte) (time.Time, []byte, error)
+}
+
+func GetCommands(ctx context.Context) (*Commands, error) {
+	sinceSupported, err := checkDmesgSupportsSinceFlag(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if sinceSupported {
+		return &Commands{
+			WatchCommands: WatchCommandsDmesg(),
+			TailCommands:  TailCommandsDmesg(),
+			ParseTimeFunc: ParseTimeFuncDmesg(),
+		}, nil
+	}
+
+	return &Commands{
+		WatchCommands: WatchCommandsJournalctl(),
+		TailCommands:  TailCommandsJournalctl(),
+		ParseTimeFunc: ParseTimeFuncJournalctl(),
+	}, nil
+}
+
+const (
+	// defaultDmesgCmd DefaultDmesgCmdWithSince default scan dmesg command (in newer util-linux it works, but older is not)
+	// some old dmesg versions don't support --since, thus fall back to the one without --since and tail the last 200 lines
+	// ref. https://github.com/leptonai/gpud/issues/32
+	defaultDmesgCmd          = "dmesg --time-format=iso --nopager --buffer-size 163920"
```

Comment:
let us bump the buffer size to 4MB (make sure we test this)... some machine may have configured a lager buffer size so we need a larger buffer here to get all messages.